chain/chain/src/test_utils.rs

```diff
@@ -1400,13 +1404,84 @@ impl ChainGenesis {
     }
 }
 
+/// A wrapper function around process_block that doesn't trigger all the callbacks
+/// Only used in tests
+pub fn process_block_test(
+    chain: &mut Chain,
+    me: &Option<AccountId>,
+    block: Block,
+) -> Result<Option<Tip>, Error> {
+    chain.process_block(
+        me,
+        MaybeValidated::from(block),
+        Provenance::PRODUCED,
+        |_| {},
+        |_| {},
+        |_| {},
+    )
+}
+
+use rand_chacha::rand_core::SeedableRng;
+use rand_chacha::ChaCha8Rng;
+use rand_distr::{Distribution, Exp};
+
+pub fn fill_mock_clock_utc(
+    mock_date: chrono::DateTime<Utc>,
+    num_samples: u64,
+    average_interval_in_miliseconds: f64,
+    seed: u64,
+) {
+    let exp = Exp::new(average_interval_in_miliseconds).unwrap();
+    let mut rng = ChaCha8Rng::seed_from_u64(seed);
+    let mut utc = mock_date.clone();
+    for _i in 0..num_samples {
+        let f = exp.sample(&mut rng) as i64;
+        let d = Duration::milliseconds(f);
+        if let Some(t) = utc.checked_add_signed(d) {
+            Clock::add_utc(utc.clone());
+            utc = t;
+        }
+    }
+}
+
+pub fn build_chain_from_random_data(data: &[u8]) {
+    if data.len() > 8 {
```

Comment:
so if data.len is smaller than 8, the test does nothing (and 'passes')?

how about we do the assert instead? (this way if someone passes a small data package, at least they get the error)